### ISSUE TITLE
Sudden failures to refresh images from Azure ACR

### ISSUE THREAD

**Issue by andrei-dascalu** [fluxcd/flux#3114]
**Describe the bug**

Starting today with no apparent changes, my flux deployment has started displaying the following error

```
component=warmer canonical_name=<acr_name>.azurecr.io/<repo_name> auth={map[]} err="requesting tags: Get https://<acr_name>.azurecr.io/v2/<repo_name>/tags/list: unauthorized: Invalid clientid or client secret."
```

**To Reproduce**

Steps to reproduce the behaviour:

1. Installed with fluxctl as follow:
```
fluxctl install \
  --git-branch master \
  --git-path app \
  --git-email flux@<company>.com \
  --git-url <bitbucket manifests repo> \
  --git-user ems-flux  > new.install.yml
```

Edited the flux manifest to add volume for azure kubernetes json

```
      - hostPath:
          path: /etc/kubernetes/azure.json
          type: File
        name: azure-config
```

added mount in flux container

```
        - mountPath: /etc/kubernetes/azure.json
          name: azure-config
          readOnly: true
```

2. Provide a GitHub repository with Kubernetes manifests: unable to do so at the moment, but I can provide examples as needed

**Expected behavior**

Flux should be able to refresh images from ACR. 

**Logs**

The only apparent issue is `unauthorized: Invalid clientid or client secret` on a number of repositories for various deployments set to be managed via flux.

**Additional context**

- Flux version: 1.19.0
- Kubernetes version: 1.15.7
- Git provider: bitbucket
- Container registry provider: Azure

**Comment by sjwoodard** [comment]
We're seeing this abruptly as well. Builds working yesterday, failed today.

**Comment by tshak** [comment]
Are either of you using Managed Identities to auth to ACR? I ask because I've seen the same issue even outside of Flux. Microsoft Azure support was not super helpful either. I've talked to a few folks who say that Managed Identities, while being the most secure option, is still not super stable. Sometimes an image pulls right away, and sometimes it takes 1-2 minutes due to unauthorized errors which resolve themselves with no operator intervention.

**Comment by andrei-dascalu** [comment]
@tshak in my case I'm not using MI, just user/pass. Also, the issue has persisted for a day (it has occurred 3 times now).
However, I'm now positive it has nothing to do with flux itself (in one case we identified the issue as expired credentials), but for the other two occurrences we found no explanation and they resolved themselves after about a day in each case (without any chances related to flux).

I am inclined to close this.
